pkg/config/config.go

```diff
@@ -309,6 +309,76 @@ var (
 	}
 )
 
+// GetProxies returns the proxy settings from the configuration
+func GetProxies() (*Proxy, error) {
+	if proxies := Datadog.Get("proxy"); proxies != nil {
+		proxies := &Proxy{}
+		if err := Datadog.UnmarshalKey("proxy", proxies); err != nil {
+			return nil, fmt.Errorf("Could not load the proxy configuration: %s", err)
+		}
+		return proxies, nil
+	}
+	return nil, nil
+}
+
+// loadProxyFromEnv overrides the proxy settings with environment variables
+func loadProxyFromEnv() {
+	// Viper doesn't handle mixing nested variables from files and set
+	// manually.  If we manually set one of the sub value for "proxy" all
+	// other values from the conf file will be shadowed when using
+	// 'Datadog.Get("proxy")'. For that reason we first get the value from
+	// the conf files, overwrite them with the env variables and reset
+	// everything.
+
+	getEnvCaseInsensitive := func(key string) string {
+		value, found := os.LookupEnv(key)
+		if found {
+			return value
+		}
+		return os.Getenv(strings.ToLower(key))
+	}
+
+	var isSet bool
+	proxies := &Proxy{}
+	if isSet = Datadog.IsSet("proxy"); isSet {
+		if err := Datadog.UnmarshalKey("proxy", proxies); err != nil {
+			isSet = false
+			log.Errorf("Could not load proxy setting from the configuration (ignoring): %s", err)
+		}
+	}
+
+	if HTTP := getEnvCaseInsensitive("HTTP_PROXY"); HTTP != "" {
+		isSet = true
+		proxies.HTTP = HTTP
+	}
+	if HTTPS := getEnvCaseInsensitive("HTTPS_PROXY"); HTTPS != "" {
+		isSet = true
+		proxies.HTTPS = HTTPS
+	}
+	if noProxy := getEnvCaseInsensitive("NO_PROXY"); noProxy != "" {
+		isSet = true
+		proxies.NoProxy = strings.Split(noProxy, ",")
```

Comment:
For other envvars, viper handles `[]string` as space separated fields, for consistency, we might want to do that too? We could even call `cast.ToStringSliceE`, that viper uses under the hood: https://github.com/spf13/cast/blob/8965335b8c7107321228e3e3702cab9832751bac/caste.go#L1052